src/GraphQL.Tests/Attributes/ParserAttributeTests.cs

```diff
@@ -185,21 +188,26 @@ public async Task parser_works_for_input_fields()
         queryType.Field<StringGraphType>("hello3")
             .Argument<AutoRegisteringInputObjectGraphType<FieldTests>>("value")
             .Resolve(ctx => ctx.GetArgument<FieldTests>("value").Field3);
+        queryType.Field<StringGraphType>("hello4")
+            .Argument<AutoRegisteringInputObjectGraphType<FieldTests>>("value")
+            .Resolve(ctx => ctx.GetArgument<FieldTests>("value").Field4);
```

Comment:
I got a strange exception before fixing the code to support private methods. Is that expected?
```
System.Exception
Failed to call Activator.CreateInstance. Type: GraphQL.Types.AutoRegisteringInputObjectGraphType`1[[GraphQL.Tests.Attributes.ParserAttributeTests+FieldTests, GraphQL.Tests, Version=8.0.1.0, Culture=neutral, PublicKeyToken=e6238258560628ee]]
   at GraphQL.DefaultServiceProvider.GetService(Type serviceType) in C:\github\graphql-dotnet\src\GraphQL\DI\DefaultServiceProvider.cs:line 32
   at GraphQL.Types.SchemaTypes.<>c__DisplayClass14_0.<Initialize>b__2(Type serviceType) in C:\github\graphql-dotnet\src\GraphQL\Types\Collections\SchemaTypes.cs:line 271
   at GraphQL.Types.SchemaTypes.AddTypeIfNotRegistered(Type type, TypeCollectionContext context) in C:\github\graphql-dotnet\src\GraphQL\Types\Collections\SchemaTypes.cs:line 769
   at GraphQL.Types.SchemaTypes.HandleField(IComplexGraphType parentType, FieldType field, TypeCollectionContext context, Boolean applyNameConverter) in C:\github\graphql-dotnet\src\GraphQL\Types\Collections\SchemaTypes.cs:line 699
   at GraphQL.Types.SchemaTypes.AddType(IGraphType type, TypeCollectionContext context) in C:\github\graphql-dotnet\src\GraphQL\Types\Collections\SchemaTypes.cs:line 555
   at GraphQL.Types.SchemaTypes.AddTypeIfNotRegistered(IGraphType type, TypeCollectionContext context) in C:\github\graphql-dotnet\src\GraphQL\Types\Collections\SchemaTypes.cs:line 785
   at GraphQL.Types.SchemaTypes.Initialize(ISchema schema, IServiceProvider serviceProvider, IEnumerable`1 graphTypeMappings) in C:\github\graphql-dotnet\src\GraphQL\Types\Collections\SchemaTypes.cs:line 294
   at GraphQL.Types.SchemaTypes..ctor(ISchema schema, IServiceProvider serviceProvider, IEnumerable`1 graphTypeMappings) in C:\github\graphql-dotnet\src\GraphQL\Types\Collections\SchemaTypes.cs:line 189
   at GraphQL.Types.SchemaTypes..ctor(ISchema schema, IServiceProvider serviceProvider) in C:\github\graphql-dotnet\src\GraphQL\Types\Collections\SchemaTypes.cs:line 174
   at GraphQL.Types.Schema.CreateSchemaTypes() in C:\github\graphql-dotnet\src\GraphQL\Types\Schema.cs:line 478
   at GraphQL.Types.Schema.CreateAndInitializeSchemaTypes() in C:\github\graphql-dotnet\src\GraphQL\Types\Schema.cs:line 449
   at GraphQL.Types.Schema.Initialize() in C:\github\graphql-dotnet\src\GraphQL\Types\Schema.cs:line 197
   at GraphQL.Tests.Attributes.ParserAttributeTests.parser_works_for_input_fields() in C:\github\graphql-dotnet\src\GraphQL.Tests\Attributes\ParserAttributeTests.cs:line 195
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass46_0.<<InvokeTestMethodAsync>b__1>d.MoveNext() in /_/src/xunit.execution/Sdk/Frameworks/Runners/TestInvoker.cs:line 253
--- End of stack trace from previous location ---
   at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in /_/src/xunit.execution/Sdk/Frameworks/ExecutionTimer.cs:line 48
   at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in /_/src/xunit.core/Sdk/ExceptionAggregator.cs:line 90

System.Reflection.TargetInvocationException
Exception has been thrown by the target of an invocation.
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at System.Activator.CreateInstance(Type type)
   at GraphQL.DefaultServiceProvider.GetService(Type serviceType) in C:\github\graphql-dotnet\src\GraphQL\DI\DefaultServiceProvider.cs:line 28

System.InvalidOperationException
Could not find method 'ParseHelloArgument' on CLR type 'FieldTests' while initializing 'FieldTests.Field4'. The method must have a single parameter of type object.
   at GraphQL.ParserAttribute.Modify(FieldType fieldType, Boolean isInputType, IGraphType graphType, MemberInfo memberInfo, Boolean& ignore) in C:\github\graphql-dotnet\src\GraphQL\Attributes\ParserAttribute.cs:line 68
   at GraphQL.Types.AutoRegisteringHelper.ApplyFieldAttributes(IGraphType graphType, MemberInfo memberInfo, FieldType fieldType, Boolean isInputType, Boolean& ignore) in C:\github\graphql-dotnet\src\GraphQL\Types\Composite\AutoRegisteringHelper.cs:line 203
   at GraphQL.Types.AutoRegisteringHelper.CreateField(IGraphType graphType, MemberInfo memberInfo, Func`2 getTypeInformation, Action`2 buildFieldType, Boolean isInputType) in C:\github\graphql-dotnet\src\GraphQL\Types\Composite\AutoRegisteringHelper.cs:line 156
   at GraphQL.Types.AutoRegisteringInputObjectGraphType`1.CreateField(MemberInfo memberInfo) in C:\github\graphql-dotnet\src\GraphQL\Types\Composite\AutoRegisteringInputObjectGraphType.cs:line 95
   at GraphQL.Types.AutoRegisteringHelper.ProvideFields(IEnumerable`1 members, Func`2 createField, Boolean isInputType)+MoveNext() in C:\github\graphql-dotnet\src\GraphQL\Types\Composite\AutoRegisteringHelper.cs:line 264
   at GraphQL.Types.AutoRegisteringInputObjectGraphType`1..ctor(AutoRegisteringInputObjectGraphType`1 cloneFrom, Expression`1[] excludedProperties, Boolean cache) in C:\github\graphql-dotnet\src\GraphQL\Types\Composite\AutoRegisteringInputObjectGraphType.cs:line 57
   at GraphQL.Types.AutoRegisteringInputObjectGraphType`1..ctor(Expression`1[] excludedProperties) in C:\github\graphql-dotnet\src\GraphQL\Types\Composite\AutoRegisteringInputObjectGraphType.cs:line 38
   at GraphQL.Types.AutoRegisteringInputObjectGraphType`1..ctor() in C:\github\graphql-dotnet\src\GraphQL\Types\Composite\AutoRegisteringInputObjectGraphType.cs:line 31
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
```

Reply:
Because DefaultServiceProvider simply uses Activator.CreateInstance, exceptions in the constructor get wrapped.